File: frontend/src/domain/valuesobject/Name.ts
export class Name {
  private constructor(private name: string) {}

  public static create(name: string): Name {
    this.validate(name);
    return new Name(name.trim());
  }

  private static validate(name: string): void {
    if (!name || typeof name !== 'string' || name.trim().length === 0) {
      throw new Error('nome é obrigatorio');
    }
  }
  public getValue(): string {
    return this.name;
  }
}

File: backend/src/domain/valuesobject/Name.go
package valuesobject

import (
	"errors"
	"strings"
)

type Name struct {
	name string
}

func NewName(name string) (*Name, error) {
	trimmed := strings.TrimSpace(name)
	if err := validateName(name); err != nil {
		return nil, err
	}
	return &Name{name: trimmed}, nil
}

func (n *Name) GetValue() string {
	return n.name
}

func validateName(name string) error {
	if len(name) == 0 {
		return errors.New("nome obrigatorio")
	}

	// isValid := regexp.MustCompile(`^[A-Za-zÀ-ÖØ-öø-ÿ]+(?: [A-Za-zÀ-ÖØ-öø-ÿ]+)*$`)

	// if !isValid.MatchString(name) {
	// 	return errors.New("o nome deve conter apenas letras e espaços sem números ou símbolos")
	// }

	return nil
}
